modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/package.scala

```diff
@@ -25,8 +25,8 @@ package object fs2 {
   type ByteSink[F[_]] = List[Array[Byte]] => F[Unit]
   type AttributedByteSink[F[_]] = List[AttributedData[Array[Byte]]] => F[Unit]
 
-  /** Enrichment result, containing list of (valid and invalid) results as well as the collector timestamp */
-  type Result = (List[Validated[BadRow, EnrichedEvent]], Option[Long])
+  type Enriched = Either[(BadRow, Option[EnrichedEvent]), EnrichedEvent]
```

Comment:
So this can be either just a BadRow or a just an EnrichedEvent or both.  Have you ever come across [cats Ior](https://typelevel.org/cats/datatypes/ior.html)?

Reply:
I didn't know about `Ior`. It seems like a good fit indeed. But what I don't like is that in case of an incomplete event, it is at the exact same place as a valid enriched event. I kind of like that the incomplete event is in the `Left`, coupled with the `BadRow`. So I'd rather leave it like that, except if you strongly think that we should swich to `Ior`.